src/ocean/text/arguments/TimeIntervalArgs.d

```diff
@@ -457,54 +621,266 @@ unittest
+    Test.test!("==")(isRelativeTime("2d/d"), true);
+
+    Test.test!("==")(isRelativeTime("1x"), false);
+    Test.test!("==")(isRelativeTime("2xm"), false);
+}
+
+/******************************************************************************
+
+    Check if a char value is a valid time unit
+
+    Params:
+        unit = the char to be checked
+
+    Returns:
+        true if the unit is valid
+
+******************************************************************************/
+
+private bool isValidTimeUnit ( char unit )
+{
+    return unit == 's' || unit == 'm' || unit == 'h' || unit == 'd';
+}
+
+/// Check if the unit is valid
+unittest
+{
+    Test.test!("==")(isValidTimeUnit('s'), true);
+    Test.test!("==")(isValidTimeUnit('m'), true);
+    Test.test!("==")(isValidTimeUnit('h'), true);
+    Test.test!("==")(isValidTimeUnit('d'), true);
+
+    size_t valid_units;
+    for ( char ch = 'a'; ch <= 'z'; ch++ )
+    {
+        if ( isValidTimeUnit(ch) )
+        {
+            valid_units++;
+        }
+    }
+
+    Test.test!("==")(valid_units, 4);
+}
+
+/******************************************************************************
+
+    Checks if a string contains a valid divided relative time
+
+    Params:
+        value = the relative time that will be validated
+
+    Returns:
+        true, if the string matches the pattern [some value]/[time unit]
+
+******************************************************************************/
+
+private bool isDividedTime ( cstring value )
+{
+    if ( value.length <= 2 )
+    {
+        return false;
+    }
 
-    if ( !toLong(value[0..$-1], tempValue) )
+    auto last_index = value.length - 1;
+
+    if ( !isValidTimeUnit(value[last_index]) )
     {
         return false;
     }
 
-    if ( unit != 's' &&  unit != 'm' && unit != 'h' && unit != 'd' )
+    if ( value[last_index - 1] != '/' )
     {
         return false;
     }
 
     return true;
 }
 
-///
+/// Validate divided time arguments
+unittest
+{
+    Test.test!("==")(isDividedTime(""), false);
+    Test.test!("==")(isDividedTime("d"), false);
+    Test.test!("==")(isDividedTime("1d"), false);
+    Test.test!("==")(isDividedTime("11d/dd"), false);
+    Test.test!("==")(isDividedTime("1d/x"), false);
+
+    Test.test!("==")(isDividedTime("11/d"), true);
+    Test.test!("==")(isDividedTime("d/d"), true);
+    Test.test!("==")(isDividedTime("11dd/d"), true);
+    Test.test!("==")(isDividedTime("1d/d"), true);
+}
+
+/******************************************************************************
+
+    Remove the time division from the relative time
+
+    Params:
+        value = the relative time
+
+    Returns:
+        the relative time without the division
+
+******************************************************************************/
+
+private cstring removeTimeDivision ( cstring value )
+{
+    if ( !isDividedTime(value) )
+    {
+        return value;
+    }
+
+    return value[0..$-2];
```

Comment:
Should we do a check on array length here?

Reply:
not really, but I added one.